app/components/primer/alpha/dialog.pcss

```diff
@@ -0,0 +1,484 @@
+    }
+
+    & .Overlay-titleWrap {
+      display: flex;
+      padding: calc(var(--primer-stack-gap-condensed, 8px) * 0.75) 0 calc(var(--primer-stack-gap-condensed, 8px) * 0.75) var(--primer-stack-gap-condensed, 8px);
+      flex-direction: column;
+      flex-grow: 1;
+      gap: var(--primer-control-small-gap, 4px);
+
+      & .Overlay-title {
+        margin: 0;
+        font-size: var(--primer-text-body-size-medium, 14px);
+        font-weight: var(--base-text-weight-semibold, 600);
+      }
+
+      & .Overlay-description {
+        margin: 0;
+        font-size: var(--primer-text-body-size-small, 12px);
+        font-weight: var(--base-text-weight-normal, 400);
+        color: var(--color-fg-muted);
+      }
+    }
+  }
+}
+
+/* generic body content slot */
+.Overlay-body {
+  padding: var(--primer-stack-padding-normal, 16px);
+  padding-top: 0;
+  overflow-y: auto;
+  scrollbar-width: thin;
+  font-size: var(--primer-text-body-size-medium, 14px);
+  flex-grow: 1;
+
+  &.Overlay-body--paddingCondensed {
+    padding: var(--primer-stack-padding-condensed, 8px);
+    padding-top: 0;
+  }
+
+  &.Overlay-body--paddingNone {
+    padding: 0;
+  }
+}
+
+/* generic footer slot */
+.Overlay-footer {
+  z-index: 1;
+  display: flex;
+  padding: 0 var(--primer-stack-padding-normal, 16px) var(--primer-stack-padding-normal, 16px) var(--primer-stack-padding-normal, 16px);
+  flex-direction: row;
+  flex-shrink: 0;
+  flex-wrap: wrap;
+
+  &.Overlay-footer--divided {
+    padding-top: var(--primer-stack-padding-normal, 16px);
+    box-shadow: inset 0 var(--primer-borderWidth-thin, 1px) var(--color-border-default);
+  }
+
+  &.Overlay-footer--alignStart {
+    justify-content: flex-start;
+    gap: var(--primer-stack-gap-condensed, 8px);
+  }
+
+  &.Overlay-footer--alignCenter {
+    justify-content: center;
+    gap: var(--primer-stack-gap-condensed, 8px);
+  }
+
+  &.Overlay-footer--alignEnd {
+    justify-content: flex-end;
+    gap: var(--primer-stack-gap-condensed, 8px);
+  }
+}
+
+/* TODO: replace with refactored IconButton */
+.Overlay-closeButton {
+  position: relative;
+  display: grid;
+  width: var(--base-size-32, 32px);
+  height: var(--base-size-32, 32px);
+  padding: 0;
+  color: var(--color-fg-muted);
+  cursor: pointer;
+  user-select: none;
+  background-color: transparent;
+  border: var(--primer-borderWidth-thin, 1px) solid transparent;
+  border-radius: var(--primer-borderRadius-medium, 6px);
+  transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+  transition-property: color, background-color, border-color;
+  place-content: center;
+  align-self: flex-start;
+  flex-shrink: 0;
+
+  &:hover,
+  &:focus {
+    background-color: var(--color-btn-hover-bg);
+    border: var(--primer-borderWidth-thin, 1px) solid var(--color-btn-hover-bg);
+  }
+
+  /* Override .close-button's `border: 0` that triggers a border-color transition on hover  */
+  &.close-button {
+    border: var(--primer-borderWidth-thin, 1px) solid transparent;
+  }
```

Comment:
Without this override the border would transition:

- `border-color: initial` (dark gray) -> `border-color: transparent`

https://user-images.githubusercontent.com/378023/204988472-9bc48f3c-f5bb-440a-9dc1-11e5fea491ce.mov

Might not be needed once `Primer::Beta::CloseButton` is updated to `IconButton`.